Library/Formula/nghttp2.rb

```diff
@@ -87,35 +89,66 @@ class Nghttp2 < Formula
     sha256 "3e15b416c9a2039c1a51208b2cd3bb4ffd796cd19e601b1d2657afcb77c3dc90"
   end
 
+  resource "Cython" do
+    url "https://pypi.python.org/packages/source/C/Cython/cython-0.22.tar.gz"
+    sha256 "14307e7a69af9a0d0e0024d446af7e51cc0e3e4d0dfb10d36ba837e5e5844015"
+  end if build.with?("python3")
+
   # https://github.com/tatsuhiro-t/nghttp2/issues/125
   # Upstream requested the issue closed and for users to use gcc instead.
   # Given this will actually build with Clang with cxx11, just use that.
   needs :cxx11
 
+  def python_version
+   @python_version ||= begin
+     build.with?('python3') ? [3, 4] : [2, 7]
+   end
+  end
+
+  def python
+   @python ||= begin
+     "python#{python_version.join('.')}"
+   end
+  end
+
+  patch :p1, :DATA
+
   def install
     ENV.cxx11
 
     if build.with? "docs"
-      ENV.prepend_create_path "PYTHONPATH", buildpath+"sphinx/lib/python2.7/site-packages"
+      ENV.prepend_create_path "PYTHONPATH", buildpath+"sphinx/lib/#{python}/site-packages"
       resources.each do |r|
         r.stage do
-          system "python", *Language::Python.setup_install_args(buildpath/"sphinx")
-        end
+          system python, *Language::Python.setup_install_args(buildpath/'sphinx')
+        end unless r.name == 'Cython'
       end
       ENV.prepend_path "PATH", (buildpath/"sphinx/bin")
     end
 
+    if build.with?('python3')
+       ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/#{python}/site-packages"
+       resource('Cython').stage do
+         system python, *(Language::Python.setup_install_args(libexec/"vendor") << "--install-scripts=#{libexec}/vendor/bin")
+       end
+       bin.env_script_all_files(libexec/"vendor/bin", :PYTHONPATH => ENV["PYTHONPATH"])
+    end
+
     args = %W[
       --prefix=#{prefix}
       --disable-silent-rules
       --enable-app
       --with-boost=#{Formula["boost"].opt_prefix}
       --enable-asio-lib
-      --disable-python-bindings
     ]
 
     args << "--enable-examples" if build.with? "examples"
     args << "--with-spdylay" if build.with? "spdylay"
+    if build.with?('python3')
+      args << "--enable-python-bindings" << "PYTHON=#{Formula['python3'].bin}/#{python}" << "CYTHON=#{libexec}/vendor/bin/cython"
```

Comment:
just use `"PYTHON=python3"`; the `:python3` dependency uses the python3 from PATH, which may not be a brewed python